### PR TITLE
feat(desing-system): Add new color tokens.

### PR DESCRIPTION
**Pull request by tajespasarela**
EX-5171

Add new token colors: Accent, Enable, Disable

# Pull request template
Add this new colors to tokens and to Desing System Utilities

## Motivation and context
These new colors are necessary for BSH setup.
Also refactored a little bit how the utilities classes are generated and the examples in the Design Syste view.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

### CODE REVIEW

File: packages/x-components/src/design-system/utilities/background-color.scss [hunk 1]
```diff
@@ -1,23 +1,7 @@
...
-  }
-  &--neutral-70 {
-    background-color: var(--x-color-base-neutral-70) !important;
-  }
-  &--neutral-95 {
-    background-color: var(--x-color-base-neutral-95) !important;
-  }
-  &--neutral-100 {
-    background-color: var(--x-color-base-neutral-100) !important;
+@import '../base/colors.tokens';
```
Comment (javieri-empathy) [inline packages/x-components/src/design-system/utilities/background-color.scss:1]: While I love the idea of automating this (finally), importing token files from other files will generate the variables each time it is imported. Extracting the size and color maps to files which only contain pure SCSS variables should be enough.

image
image
image

File: packages/x-components/src/design-system/utilities/box-shadow.scss [hunk 2]
```diff
@@ -4,7 +4,7 @@
   box-shadow: none !important;
 }
 .x-shadow {
-  @for $i from 1 through 10 {
+  @for $i from 0 through 10 {
```
Comment (javieri-empathy) [inline packages/x-components/src/design-system/utilities/box-shadow.scss:7]: Why did you add a `shadow-0`? you have a `shadow-none` just above

File: packages/x-components/src/views/design-system/design-system-utilities.vue [hunk 3]
```diff
@@ -504,7 +345,41 @@
...
+      'enable',
+      'disable'
+    ];
+
+    protected maxSize = 20;
+
+    protected get sizes(): string[] {
+      return Array.from({ length: this.maxSize + 1 }).map((_, index) =>
+        String(index).padStart(2, '0')
+      );
```
Comment (javieri-empathy) [inline packages/x-components/src/views/design-system/design-system-utilities.vue:367]: Just FYI, you can pass the `map` as the second `Array.from` parameter:

```suggestion
      return Array.from({ length: this.maxSize + 1 }, (_, index) => String(index).padStart(2, '0'));
```

File: packages/x-components/src/views/design-system/design-system-utilities.vue [hunk 4]
```diff
@@ -504,7 +345,41 @@
   import { Component } from 'vue-property-decorator';
 
   @Component
-  export default class DesignSystemUtilities extends Vue {}
+  export default class DesignSystemUtilities extends Vue {
```
Comment (javieri-empathy) [inline packages/x-components/src/views/design-system/design-system-utilities.vue:348]: On behalf of my CPU, thanks

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/views/design-system/design-system-utilities.vue (hunk 3 above)
Comment (tajespasarela) [reply packages/x-components/src/views/design-system/design-system-utilities.vue]: wonderful! thanks

File: packages/x-components/src/views/design-system/design-system-utilities.vue (hunk 4 above)
Comment (tajespasarela) [reply packages/x-components/src/views/design-system/design-system-utilities.vue:348]: little step by little step

File: packages/x-components/src/design-system/utilities/box-shadow.scss (hunk 2 above)
Comment (tajespasarela) [reply packages/x-components/src/design-system/utilities/box-shadow.scss]: Didn't realize, changed

**Review by javieri-empathy — APPROVED** [review]

File: packages/x-components/src/design-system/utilities/background-color.scss (hunk 1 above)
Comment (javieri-empathy) [reply packages/x-components/src/design-system/utilities/background-color.scss]: Even if I don't know how I managed to make `@use` work, it still doesn't help us. If we have our variables or mixins defined in a file which renders real CSS content, `rollup-plugin-styles` will duplicate these styles as many times as they are imported.

**Review by pmareke — APPROVED** [review]
I'm not sure if I'm understanding right but why the title accent color is different?

image

Should we add in the Typography also the enable and disabled?

image

**Review by pmareke — CHANGES_REQUESTED** [review]
Change accent color in typography and add missing enable and disable.

**Review by pmareke — APPROVED** [review]
